modules/platforms/cpp/thin-client/include/ignite/thin/cache/query/query_scan.h

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * @file
+ * Declares ignite::thin::cache::query::ScanQuery class.
+ */
+
+#ifndef _IGNITE_THIN_CACHE_QUERY_QUERY_SCAN
+#define _IGNITE_THIN_CACHE_QUERY_QUERY_SCAN
+
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+#include <ignite/impl/thin/copyable_writable.h>
+
+namespace ignite
+{
+    namespace impl
+    {
+        namespace thin
+        {
+            // Forward declaration
+            class ScanQueryRequest;
+        }
+    }
+
+    namespace thin
+    {
+        namespace cache
+        {
+            namespace query
+            {
+               /**
+                * Scan query.
+                */
+                class ScanQuery
+                {
+                public:
+                    friend class ScanQueryRequest;
+
+                    /**
+                     * Default constructor.
+                     */
+                    ScanQuery() : part(-1), pageSize(1024), loc(false)
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Constructor.
+                     *
+                     * @param part Partition.
+                     */
+                    ScanQuery(int32_t part) : part(part), pageSize(1024), loc(false)
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Get partition to scan.
+                     *
+                     * @return Partition to scan.
+                     */
+                    int32_t GetPartition() const
+                    {
+                        return part;
+                    }
+
+                    /**
+                     * Set partition to scan.
+                     *
+                     * @param partition Partition to scan.
+                     */
+                    void SetPartition(int32_t partition)
+                    {
+                        this->part = partition;
+                    }
+
+                    /**
+                     * Get page size.
+                     *
+                     * @return Page size.
+                     */
+                    int32_t GetPageSize() const
+                    {
+                        return pageSize;
+                    }
+
+                    /**
+                     * Set the size of the resul page.
+                     *
+                     * @param resultPageSize Result page size.
+                     */
+                    void SetPageSize(int32_t resultPageSize)
+                    {
+                        this->pageSize = resultPageSize;
+                    }
+
+                    /**
+                     * Get local flag.
+                     *
+                     * @return Local flag.
+                     */
+                    bool IsLocal() const
+                    {
+                        return loc;
+                    }
+
+                    /**
+                     * Set local flag.
```

Comment:
Should we provide more details here - what does it mean to do a local query?

Reply:
Fixed.